diff -urp a/drivers/video/mxc/mxcfb_sharp_wsvga.c b/drivers/video/mxc/mxcfb_sharp_wsvga.c
--- a/drivers/video/mxc/mxcfb_sharp_wsvga.c	2009-10-19 16:52:39.000000000 +0900
+++ b/drivers/video/mxc/mxcfb_sharp_wsvga.c	2009-10-19 16:53:22.000000000 +0900
@@ -68,7 +68,7 @@ static void lcd_init_fb(struct fb_info *
 	fb_videomode_to_var(&var, &video_modes[0]);
 
 	var.activate = FB_ACTIVATE_ALL;
-	var.yres_virtual = var.yres * 2;
+	var.yres_virtual = var.yres;
 
 	acquire_console_sem();
 	info->flags |= FBINFO_MISC_USEREVENT;
